### PR TITLE
Migrate to Babel 7

### PR DESCRIPTION
**Pull request by tylerhou**

### CODE REVIEW

File: src/elements/ModuleSpecifier.js [hunk 1]
```diff
@@ -12,6 +12,4 @@ export default class ModuleSpecifier extends Node {
         super(type, children);
         this.isModuleSpecifier = true;
     }
-
-    local: Object;
```
Comment (tylerhou) [inline src/elements/ModuleSpecifier.js:16]: If I don't remove this line `moduleSpecifier.local` gets set to `undefined` after the constructor runs

File: src/elements/types/ObjectExpression.js [hunk 2]
```diff
@@ -17,8 +17,8 @@ export default class ObjectExpression extends Expression {
                 children.skipNonCode();
                 children.assertToken('Punctuator', '}');
             } else {
-                if (children.isNode('SpreadProperty')) {
-                    properties.push(children.passNode('SpreadProperty'));
+                if (children.isNode('SpreadElement')) {
+                    properties.push(children.passNode('SpreadElement'));
```
Comment (tylerhou) [inline src/elements/types/ObjectExpression.js:21]: `SpreadProperty` and `RestProperty` have been removed.

https://babeljs.io/docs/en/v7-migration-api#restproperty-spreadproperty

File: src/elementTree.js [hunk 3]
```diff
@@ -211,9 +210,8 @@ function processToken(token: Object, source: string): BabylonToken {
         type === tt.bitwiseAND || type === tt.equality ||
         type === tt.relational || type === tt.bitShift ||
         type === tt.plusMin || type === tt.modulo ||
-        type === tt.exponent || type === tt.prefix ||
-        type === tt.doubleColon ||
-        type.isAssign) {
+        type === tt.exponent || type === tt.doubleColon ||
+        type === tt.tilde || type === tt.bang || type.isAssign) {
```
Comment (tylerhou) [inline src/elementTree.js:214]: https://github.com/babel/babel/commit/97c23461f9e27f33f0af58f3148efa68894bd332

**Comment by tylerhou** [comment]
Few things:

1. Looks like Babel dropped support for Node 4. We have to a major version upgrade.
2. Didn't realize that there was a submodule for babylon fixtures. Fixing those tests right now.

**Comment by tylerhou** [comment]
Is it a goal for CST to support flow, TypeScript, and experimental JS features?

**Comment by markelog** [comment]
> Looks like Babel dropped support for Node 4. We have to a major version upgrade.

Sounds good to me

> Didn't realize that there was a submodule for babylon fixtures. Fixing those tests right now.

Oh yeah, that's our gotcha

> Is it a goal for CST to support flow, TypeScript, and experimental JS features?

I would say yes :), "how" that's another convo :).

So it seems those fixtures are still an issue?

**Comment by tylerhou** [comment]
Fixtures are a big issue. See #167; I have more details there.

**Pull request closed without merging**